### ISSUE TITLE
Resizing with no transforms results in inconsistent tensor sizes

### ISSUE THREAD

**Issue by cluePrints** [fastai/fastai#1602]
**Describe the bug**
Resizing with no transforms leads to inconsistent resulting tensor sizes within batches produced `labeled_list.transform(tfms=None, size=10)`.

As per discussion on https://github.com/fastai/fastai/issues/1494 I intend to fix this.

```
/usr/local/lib/python3.6/dist-packages/fastai/basic_data.py:236: UserWarning: It's not possible to collate samples of your dataset together in a batch.
Shapes of the inputs/targets:
[[torch.Size([3, 200, 300]), torch.Size([3, 201, 299])], [(), ()]]
  warn(message)
```

This in turn breaks the training process.

```
RuntimeError: invalid argument 0: Sizes of tensors must match except in dimension 0.
Got 203 and 204 in dimension 2 at /pytorch/aten/src/TH/generic/THTensorMoreMath.cpp:1333
```

**Provide your installation details**
```text
=== Software === 
python       : 3.6.7
fastai       : 1.0.42
fastprogress : 0.1.18
torch        : 1.0.0
torch cuda   : 10.0.130 / is **Not available** 

=== Hardware === 
No GPUs available     (Also reproducible on GPU)

=== Environment === 
platform     : Linux-4.14.79+-x86_64-with-Ubuntu-18.04-bionic
distro       : #1 SMP Wed Dec 19 21:19:13 PST 2018
conda env    : Unknown
python       : /usr/bin/python3
sys.path     : 
/env/python
/usr/lib/python36.zip
/usr/lib/python3.6
/usr/lib/python3.6/lib-dynload
/usr/local/lib/python3.6/dist-packages
/usr/lib/python3/dist-packages
/usr/local/lib/python3.6/dist-packages/IPython/extensions
/root/.ipython
no supported gpus found on this system
```

**To Reproduce**
- 1.0.41 / cpu / google colab ([notebook](https://colab.research.google.com/drive/1A9H-6QyS-NlTfZaNjCM2TbifODmvxOTG#scrollTo=_kuCPJS1r_rJ))
- [pytest](https://gist.github.com/cluePrints/cb8cde19c6094b889c8ff407f3f1ad69)

**Expected behavior**
Batches of same sized tensors produced. Training process can proceed.

**Screenshots**
```
RuntimeError: Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py", line 138, in _worker_loop
    samples = collate_fn([dataset[i] for i in batch_indices])
  File "/usr/local/lib/python3.6/dist-packages/fastai/torch_core.py", line 110, in data_collate
    return torch.utils.data.dataloader.default_collate(to_data(batch))
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py", line 232, in default_collate
    return [default_collate(samples) for samples in transposed]
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py", line 232, in <listcomp>
    return [default_collate(samples) for samples in transposed]
  File "/usr/local/lib/python3.6/dist-packages/torch/utils/data/dataloader.py", line 209, in default_collate
    return torch.stack(batch, 0, out=out)
RuntimeError: invalid argument 0: Sizes of tensors must match except in dimension 0. Got 208 and 206 in dimension 2 at /pytorch/aten/src/TH/generic/THTensorMoreMath.cpp:1333
```

**Additional context**
When multiple workers are used, this results in occasional (1/5 executions) indefinite waits (I've tried waiting 5 mins before giving up).

**Comment by stas00** [comment]
Loving your notebook! I'm sure @sgugger would love all bug reports to include one!

FYI, in 1.0.43 (or current git) we now have `URLs.MNIST_VAR_SIZE_TINY`, so there is a real dataset with variable size to test with. But I'm loving your creative use of `f'https://picsum.photos/{w}/{h}'`.

I added to your notebook:
```
!pip install git+https://github.com/fastai/fastai
```
so we test the latest code base, and reduced the problem to the `resize_method`.

it works if I use resize_method=ResizeMethod.PAD, it fails with SQUISH and CROP

For some reason the other two fail to resize to the requested dimension, getting the height right, but failing to get the width to match. Perhaps this has something to do with the type of the image? It fails to scale up in the same way.

Also this one is odd:
```
UserWarning: It's not possible to collate samples of your dataset together in a batch.
Shapes of the inputs/targets:
[[torch.Size([3, 68, 50]), torch.Size([3, 93, 50])], [(), ()]]
```
what's that odd empty image? (I changed the code to resize to 50x50)
